migration/actual_migrations_test.go

```diff
@@ -370,35 +371,132 @@ func TestMigration16(t *testing.T) {
 	db, dbDriver, closer := prepareDBAndInfo(t)
 	defer closer()
 
-	if dbDriver == types.DBDriverSQLite3 {
-		// migration is not implemented for sqlite
-		return
-	}
-
 	err := migration.SetDBVersion(db, dbDriver, 15)
 	helpers.FailOnError(t, err)
 
 	_, err = db.Exec(`
-		INSERT INTO recommendations (cluster, rule_fqdn, error_key)
-		VALUES ($1, $2, $3)
+		INSERT INTO recommendation (org_id, cluster_id, rule_fqdn, error_key)
+		VALUES ($1, $2, $3, $4)
 	`,
+		testdata.OrgID,
 		testdata.ClusterName,
 		testdata.Rule1Name,
 		testdata.ErrorKey1,
 	)
-	assert.EqualError(t, err, "no such table: recommendations")
+	assert.Error(t, err, `Expected error since recommendation table does not exist yet`)
+
+	if dbDriver == types.DBDriverSQLite3 {
+		assert.Contains(t, err.Error(), "no such table: recommendation")
+	} else if dbDriver == types.DBDriverPostgres {
+		assert.Contains(t, err.Error(), `relation "recommendation" does not exist`)
+	}
 
 	err = migration.SetDBVersion(db, dbDriver, 16)
 	helpers.FailOnError(t, err)
 
 	_, err = db.Exec(`
-		INSERT INTO recommendations (cluster, rule_fqdn, error_key)
-		VALUES ($1, $2, $3)
+		INSERT INTO recommendation (org_id, cluster_id, rule_fqdn, error_key)
+		VALUES ($1, $2, $3, $4)
 	`,
+		testdata.OrgID,
 		testdata.ClusterName,
 		testdata.Rule1Name,
 		testdata.ErrorKey1,
 	)
 	helpers.FailOnError(t, err)
+}
+
+func TestMigration19(t *testing.T) {
+
+	db, dbDriver, closer := prepareDBAndInfo(t)
+	defer closer()
+
+	if dbDriver == types.DBDriverSQLite3 {
+		// nothing worth testing for sqlite
+		return
+	}
+
+	err := migration.SetDBVersion(db, dbDriver, 18)
+	helpers.FailOnError(t, err)
+
+	correctRuleID := testdata.Rule1ID + "|" + testdata.ErrorKey1
+	incorrectRuleFQDN := testdata.Rule1ID + "." + testdata.ErrorKey1
+
+	expectedRuleAfterMigration := string(testdata.Rule1ID)
+
+	_, err = db.Exec(`
+		INSERT INTO recommendation (org_id, cluster_id, rule_fqdn, error_key)
+		VALUES ($1, $2, $3, $4)
+		`,
+		testdata.OrgID,
+		testdata.ClusterName,
+		incorrectRuleFQDN,
+		testdata.ErrorKey1,
+	)
+	helpers.FailOnError(t, err)
 
+	_, err = db.Exec(`
+		INSERT INTO recommendation (org_id, cluster_id, rule_fqdn, error_key)
+		VALUES ($1, $2, $3, $4)
+		`,
+		testdata.Org2ID,
+		testdata.ClusterName,
+		correctRuleID,
+		testdata.ErrorKey1,
+	)
+	helpers.FailOnError(t, err)
+
+	err = migration.SetDBVersion(db, dbDriver, 19)
+	helpers.FailOnError(t, err)
+
+	var (
+		ruleFQDN string
+		ruleID   string
+	)
+
+	err = db.QueryRow(`
+			SELECT
+				rule_fqdn, rule_id
```

Comment:
`error_key`?

Reply:
Not sure what's the question =)


I'm not testing `error_key` in migration 19 because at that point it already exists in the table.